### PR TITLE
[node] Nodes can use store service

### PR DESCRIPTION
**Pull request by cf19drofxots**
### Description

Given a store service, Nodes can store and retrieve data.

https://github.com/counterfactual/monorepo/blob/dc9e1e816bd70b67fa7d3588a3e725a237bf39d5/packages/node/test/integration/node-store.spec.ts#L31-L39

### CODE REVIEW

File: packages/node/test/integration/services/firebase-service.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import firebase from "firebase";
+import FirebaseServer from "firebase-server";
+
+import {
+  IMessagingService,
+  IStoreService
+} from "../../../src/service-interfaces";
+
+const MESSAGING_SERVER_KEY = "messages";
+const STORE_SERVER_KEY = "store";
```
Comment (snario) [inline packages/node/test/integration/services/firebase-service.ts]: Should pull these values from the environment and be different per deployment

File: packages/node/test/integration/services/firebase-service.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    if (!this.firebase.app) {
+      console.error(
+        "Cannot register a connection with an uninitialized firebase handle"
+      );
+      return;
+    }
+
+    this.firebase
+      .ref(`${MESSAGING_SERVER_KEY}/${address}`)
+      // The snapshot being sent to this call _might_ be null
```
Comment (snario) [inline packages/node/test/integration/services/firebase-service.ts]: why? this comment adds no information different from `| null` below

File: packages/node/test/integration/services/firebase-service.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      .on("value", (snapshot: firebase.database.DataSnapshot | null) => {
+        if (snapshot === null) {
+          console.debug(
+            `Node with address ${address} received a "null" snapshot`
+          );
+          return;
+        }
+        const msg = snapshot.val();
+        callback(msg);
```
Comment (snario) [inline packages/node/test/integration/services/firebase-service.ts]:
```suggestion
        callback(snapshot.val());
```

File: packages/node/test/integration/services/firebase-service.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      });
+    return result;
+  }
+
+  async set(key: string, value: any): Promise<any> {
+    return await this.firebase.ref(`${STORE_SERVER_KEY}/${key}`).set(value);
+  }
+}
+
+const HOST = "localhost";
```
Comment (snario) [inline packages/node/test/integration/services/firebase-service.ts]: Again store these somewhere else

File: packages/node/src/node.ts [hunk 5]
```diff
@@ -75,6 +76,27 @@ export default class Node {
...
 
+  // Note: The following getter/setter method will become private
+  // once the machine becomes embedded in the Node.
+  /**
+   * Retrieves the value that's mapped to the given key through the provided
+   * When the given key doesn't map to any value, `null` is returned.
+   * store service.
+   * @param key
+   */
+  async get(key: string): Promise<any> {
```
[the remark this answers is not in the document]
Comment (cf19drofxots) [reply packages/node/src/node.ts]: The functionality that primarily uses the store service is the machine storing commitments, which is why this PR is a prerequisite for embedding the machine.

File: packages/node/test/integration/services/firebase-service.ts (hunk 1 above)
Comment (cf19drofxots) [reply packages/node/test/integration/services/firebase-service.ts]: This is specifically under the test scope though, and so it'll be the same across deployments.

For services that are actually injected into the Node, those should probably pull this stuff from the environment.

**Pull request merged**